central/resourcecollection/datastore/datastore_impl.go

```diff
@@ -427,3 +431,161 @@ func verifyCollectionObjectNotEmpty(obj *storage.ResourceCollection) error {
+	if collection.GetResourceSelectors() != nil && len(collection.GetResourceSelectors()) > 1 {
+		return nil, errors.New("only 1 resource selector is supported per collection")
+	}
+
+	query, err := ds.resolveCollectionQuery(ctx, collection)
+	if err != nil {
+		return nil, err
+	}
+	return ds.deploymentDS.SearchListDeployments(ctx, query)
+}
+
+func (ds *datastoreImpl) resolveCollectionQuery(ctx context.Context, collection *storage.ResourceCollection) (*v1.Query, error) {
+	var collections []*storage.ResourceCollection
+	var collectionSet set.Set[string]
+	var disjunctions []*v1.Query
+
+	collections = append(collections, collection)
+
+	ds.lock.RLock()
+	defer ds.lock.RUnlock()
+
+	for len(collections) > 0 {
+
+		// get first index and remove from list
+		collection := collections[0]
+		collections = collections[1:]
+
+		if !collectionSet.Add(collection.GetId()) {
+			continue
+		}
+
+		// resolve the collection to queries
+		queries, err := collectionToQueries(collection)
+		if err != nil {
+			return nil, err
+		}
+		disjunctions = append(disjunctions, queries...)
+
+		// add embedded values
+		embeddedList, _, err := ds.storage.GetMany(ctx, embeddedCollectionsToIDList(collection.GetEmbeddedCollections()))
+		if err != nil {
+			return nil, err
+		}
+		collections = append(collections, embeddedList...)
+	}
+
+	return pkgSearch.DisjunctionQuery(disjunctions...), nil
+}
+
+// collectionToQueries returns a list of queries derived from the given resource collection's storage.ResourceSelector list
+// these should be combined as disjunct with any resolved embedded queries
+func collectionToQueries(collection *storage.ResourceCollection) ([]*v1.Query, error) {
+	var ret []*v1.Query
+
+	for _, resourceSelector := range collection.GetResourceSelectors() {
+		var selectorRuleQueries []*v1.Query
+		for _, selectorRule := range resourceSelector.GetRules() {
+
+			// currently we only support disjunction operations here
+			if selectorRule.GetOperator() != storage.BooleanOperator_OR {
+				return nil, errors.Wrap(errox.InvalidArgs, "`and` boolean operator unsupported")
+			}
+
+			fieldLabel, present := supportedFieldNames[selectorRule.GetFieldName()]
+			if !present {
+				return nil, errors.Wrapf(errox.InvalidArgs, "unsupported field label (%s)", selectorRule.GetFieldName())
+			}
+
+			ruleValueQueries := ruleValuesToQueryList(fieldLabel, selectorRule.GetValues())
+			if len(ruleValueQueries) > 0 {
+				// rule values are conjunct or disjunct with each other depending on the operator
+				switch selectorRule.GetOperator() {
+				case storage.BooleanOperator_OR:
+					selectorRuleQueries = append(selectorRuleQueries, pkgSearch.DisjunctionQuery(ruleValueQueries...))
+				case storage.BooleanOperator_AND:
+					selectorRuleQueries = append(selectorRuleQueries, pkgSearch.ConjunctionQuery(ruleValueQueries...))
+				default:
+					return nil, errors.Wrap(errox.InvalidArgs, "unsupported boolean operator")
+				}
+			}
+		}
```

Comment:
Refactor this into `validateResourceSelectors` and also add to create and update workflows. 

Additionally, put all the validations in one func, `validateCollection(forUpdate bool)` func.

Reply:
I think it would be best to get this functionality in before looking at doing this refactor